caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -48,18 +48,14 @@ export function setFilterColumnOpts(filterColumnOpts) {
 }
 
 export function resetFormData() {
-  // Clear all form data when switching datasource
+  // Clear all form data when switching datasource or viz_type
   return { type: RESET_FORM_DATA };
```

Comment:
When user switches datasource we would reset both choices and already selected values in all forms. I'm thinking about separating out actions for onChange of datasource/viz_type in another PR, and dispatch resetFormData and resetChoices from there.